docs/classes/response.md

```diff
@@ -3,92 +3,113 @@ title: Response
 permalink: /classes/response
 ---
 
-{:.warning}
-> This page needs an update.
-
 * Table of Contents
 {:toc}
 
-The `Response` interface (extends `\Amp\ByteStream\OutputStream`) generally finds its only use in responder callables (or [`Websocket::onOpen()`](websocket.md#onopenint-clientid-handshakedata)). [`Middleware`s](middleware.md) do never see the `Response`; the `StandardResponse` class is communicating headers, data and flushes to a Generator under the hood.
+The **`Response`** class represents an HTTP response. The **`Response`** promise is returned by [request handlers](request-handler.md) and [`middleware`](middleware.md).
 
-## `setStatus(int $code): Response`
+## Constructor
 
-Sets the numeric HTTP status code (between 100 and 599).
+```php
+public function __construct(
+    int $code = Status::OK, 
+    string[] | string[][] $headers = [], 
+    Amp\ByteStream\InputStream | string | null $stringOrStream = null
+)
+```
 
-If not assigned this value defaults to 200.
+### Parameters
 
-## `setReason(string $phrase): Response`
+|`int`|`$code`|HTTP response status code|
+|`string[]`<br />`string[][]`|`$headers`|An array of strings or an array of string arrays.|
+|[`Amp\ByteStream\InputStream`](https://amphp.org/byte-stream/)<br />`string`<br />`null`|`$stringOrStream`|Response body|
 
-Sets the optional HTTP reason phrase.
+## Destructor
 
-## `addHeader(string $field, string $value): Response`
+Invokes dispose handlers (i.e. functions that registered via [`onDispose`](#ondisposecallable-ondispose) method).
 
-Appends the specified header.
+{:.note}
+> Uncaught exceptions from the dispose handlers will be forwarded to the [event loop](https://amphp.org/amp/event-loop/) error handler.
 
-## `setHeader(string $field, string $value): Response`
+## `getBody(): Amp\ByteStream\InputStream`
 
-Sets the specified header.
+Returns the [stream](https://amphp.org/byte-stream/) for the message body.
 
-This method will replace any existing headers for the specified field.
+## `setBody(\Amp\ByteStream\InputStream | string | null $stringOrStream)`
 
-## `setCookie(string $name, string $value, array $flags = []): Response`
+Sets the [stream](https://amphp.org/byte-stream/) for the message body.
 
-Provides an easy API to set cookie headers.
+{:.note}
+> Using a string will automatically set the `Content-Length` header to the length of the given string. Setting a [`stream`](https://amphp.org/byte-stream/) will remove the `Content-Length` header.
```

Comment:
Please add something like: _"If you know the exact content length of your stream, you can add a `content-length` header after calling `setBody()`."_